### ISSUE TITLE
Missing `@babel/core` dependency

### ISSUE THREAD

**Issue by rtritto** [vercel/styled-jsx#736]
# Summary

After I install `styled-jsx` with yarn, I get dependency not found warnings in yarn log.

`@babel/core` should be added in dependencies.

# Versions

- `styled-jsx`: 4.0.0

# How To Reproduce

1. `yarn init -y`
2. `yarn set version berry`
3. `yarn add styled-jsx react`
4. See yarn log

# Results
## Actual
```
➤ YN0000: ┌ Resolution step
➤ YN0002: │ styled-jsx@npm:4.0.0 [2b948] doesn't provide @babel/core (pfa7cf), requested by @babel/plugin-syntax-jsx
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ string-hash@npm:1.1.3 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ styled-jsx@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ stylis-rule-sheet@npm:0.0.10 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ stylis@npm:3.5.4 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ to-fast-properties@npm:2.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 0s 565ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: Done with warnings
```

## Expected
No warning in yarn log.

# Workaround
Add in `.yarnrc.yml`:
```YML
packageExtensions:
  styled-jsx@*:
    dependencies:
      "@babel/core": "*"
```

**Comment by huozhi** [comment]
Tested the error is gone with `next@11.1.2` on my side

**Comment by huozhi** [comment]
Resolved by #32742 , you can install latest canary (`next@12.0.9-canary.2`) to get rid of this warning
